source/engine/engine/ComponentFactories.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ComponentFactories.h"
+#include "ncengine/asset/DefaultAssets.h"
+#include "ncengine/audio/AudioSource.h"
+#include "ncengine/ecs/Logic.h"
+#include "ncengine/ecs/Registry.h"
+#include "ncengine/graphics/ParticleEmitter.h"
+#include "ncengine/graphics/MeshRenderer.h"
+#include "ncengine/graphics/ToonRenderer.h"
+#include "ncengine/network/NetworkDispatcher.h"
+#include "ncengine/physics/Collider.h"
+#include "ncengine/physics/PhysicsBody.h"
+
+namespace nc
+{
+auto CreateAudioSource(Entity entity, void*) -> audio::AudioSource
```

Comment:
I am not a huge fan of having the majority of these methods accept a parameter they don't use. If it's an acceptable performance hit, I'd much rather they accept the param, and propagate if not null. Then a user would not have to make changes to the engine to pass their own userdata in

Reply:
discussed